alignment/rule_based_detector.py

```diff
@@ -0,0 +1,182 @@
+import re
+from difflib import SequenceMatcher
+from collections import namedtuple
+
+import nltk
+
+from text_segmenter import HardLineBreakDetector
+
+CHINESE_NUM_DICT = {'零': 0, '一': 1, '二': 2, '三': 3, '四': 4,
+                '五': 5, '六': 6, '七': 7, '八': 8, '九': 9}
+CHINESE_UNIT_DICT = {'十': 10, '百': 100, '千': 1000, '万': 10000, '亿': 100000000}
+
+ROMAN_VAL = {
+    'I': 1,
+    'V': 5,
+    'X': 10,
+    # 'L': 50,
+}
+
+def read_chinese(s: str) -> int:
+    """读汉字"""
+    num = 0
+    unit = 1
+    for digit in reversed(s):
+        if digit in CHINESE_UNIT_DICT:
+            if CHINESE_UNIT_DICT[digit] < unit:
+                unit = CHINESE_UNIT_DICT[digit]
+                num += unit
+            else:
+                unit = CHINESE_UNIT_DICT[digit]
+        elif digit in CHINESE_NUM_DICT:
+            num += CHINESE_NUM_DICT[digit] * unit
+    return num
+
+def read_int(s: str) -> int:
```

Comment:
you can do return int(s)

Reply:
`read_int` is designed for reading a int from a string robustly (so we can use it under multiple cases and combine the rule with the unordered list rules), or we may use more complex logic for LINENO_SEG_TOKENS match.